jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -258,10 +271,49 @@ public BaseImageWithAuthorization call()
           ContainerConfigurationTemplate containerConfigurationTemplate =
               JsonTemplateMapper.readJson(
                   containerConfigurationString, ContainerConfigurationTemplate.class);
-          return JsonToImageTranslator.toImage(v22ManifestTemplate, containerConfigurationTemplate);
+          buildConfiguration
+              .getBaseImageLayersCache()
+              .writeMetadata(
+                  buildConfiguration.getBaseImageConfiguration().getImage(),
+                  buildableManifestTemplate,
+                  containerConfigurationTemplate);
+          return JsonToImageTranslator.toImage(
+              buildableManifestTemplate, containerConfigurationTemplate);
         }
     }
 
     throw new IllegalStateException("Unknown manifest schema version");
   }
+
+  /**
+   * Retrieves the cached base image.
+   *
+   * @return the cached image
+   * @throws IOException when an I/O exception occurs
+   * @throws CacheCorruptedException if the cache is corrupted
+   * @throws LayerPropertyNotFoundException if adding image layers fails
+   * @throws BadContainerConfigurationFormatException if the container configuration is in a bad
+   *     format
+   */
+  private Image<Layer> pullBaseImageOffline()
+      throws IOException, CacheCorruptedException, BadContainerConfigurationFormatException,
+          LayerCountMismatchException {
+    ImageReference baseImage = buildConfiguration.getBaseImageConfiguration().getImage();
+    Optional<ManifestAndConfig> metadata =
+        buildConfiguration.getBaseImageLayersCache().retrieveMetadata(baseImage);
+    if (!metadata.isPresent()) {
+      throw new IOException(
+          "Cannot run Jib in offline mode; " + baseImage + " not found in local Jib cache");
+    }
+
+    ManifestTemplate manifestTemplate = metadata.get().getManifest();
+    if (manifestTemplate instanceof V21ManifestTemplate) {
+      return JsonToImageTranslator.toImage((V21ManifestTemplate) manifestTemplate);
+    }
+
+    ContainerConfigurationTemplate configurationTemplate =
+        metadata.get().getConfig().orElseThrow(IllegalStateException::new);
```

Comment:
I'm fine with `metadata.get().getConfig().get()`. Not much different from throwing `NoSuchElementException`. This should never happen, unless there is a bug.

Reply:
Right, I only added this to avoid an IntelliJ warning.